pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -177,20 +178,33 @@ func (c *Reconciler) reconcile(ctx context.Context, pa *pav1alpha1.PodAutoscaler
 	// Compare the desired and observed resources to determine our situation.
 	// We fetch private endpoints here, since for scaling we're interested in the actual
 	// state of the deployment.
-	got := 0
+	ready, notReady, pending, terminating := 0, 0, 0, 0
```

Comment:
> If sks is not ready or the endpoint counter returns an error, we will report 0 for notReady/pending/terminating. Can that be potentially misleading?

if there is an error from the endpoints counter or the pod counter, we `return` with an error and wont report anything. Otherwise, zero seems to be the sensible value to report here.

> nit: can also we wrap these in a struct, and pass the got object into computeStatus?

yeah possibly. the struct doesn't buy us much though since we are not using it anywhere else. so not sure if it is of much value.

Reply:
So this will only report anything on the autoscaler suggestion change.
Or on global resync.
Which means we can have a node migration with 10pods terminating and coming to life and we'll never even see those events.
Should this run in its own timer based loop elsewhere?

Reply:
yeah, it is meant to be used when making autoscaling decisions, similar to how we use actual and desired pod counts. The grafana dashboard that plots these metrics is also under `Knative Serving - Scaling Debugging`.  I don't see it as necessarily relevant information for end users.